sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1181,7 +1181,9 @@ class Column(val expr: Expression) extends Logging {
    * @group expr_ops
    * @since 1.3.0
    */
-  def cast(to: DataType): Column = withExpr { Cast(expr, to) }
+  def cast(to: DataType): Column = withExpr {
+    Cast(expr, CharVarcharUtils.replaceCharVarcharWithString(to))
+  }
```

Comment:
So we can do `cast(CharType)`? It actually casts to StringType? But don't we loss length info?

Reply:
If you do `col.cast("char(5)")` before this PR, Spark already silently treats it as cast to string type. I don't want to change this behavior here. We can make it better later, by failing explicitly and saying that cast to char type is not supported.